### PR TITLE
[init] Add option for reusing database

### PR DESCRIPTION
**Pull request by jgbarah**
Really, for raising a new exception when intending to create a new database (using init). And a new option to Init, reuse, that will not trigger printing the message about "error creating database" which appears now every time you call Init.

### CODE REVIEW

File: tests/test_cmd_init.py [hunk 1]
```diff
@@ -114,14 +113,23 @@ def test_existing_db_error(self):
...
+        self.assertEqual(code2, CODE_DATABASE_EXISTS)
 
         # Context added to catch deprecation warnings raised on Python 3
         with warnings.catch_warnings(record=True):
             output = sys.stderr.getvalue().strip()
             self.assertRegexpMatches(output,
                                      DB_EXISTS_ERROR % {'database' : self.name})
 
+    def test_existing_db_reuse(self):
+        """Check if it returns an error when tries to create the registry twice"""
```
Comment (sduenas) [inline tests/test_cmd_init.py]: The description of this test is wrong. Calling to run doesn't return an error in any of the cases.

File: tests/test_cmd_init.py [hunk 2]
```diff
@@ -166,14 +175,31 @@ def test_existing_db_error(self):
...
             self.assertRegexpMatches(output,
                                      DB_EXISTS_ERROR % {'database' : self.name})
 
+        cmd = Init(user=self.kwargs['user'], password=self.kwargs['password'],
+                    host=self.kwargs['host'], port=self.kwargs['port'])
+        code3 = cmd.initialize(self.name)
+        self.assertEqual(code2, CODE_DATABASE_EXISTS)
+
+    def test_existing_db_reuse(self):
+        """Check if it returns an error when tries to create the registry twice"""
```
Comment (sduenas) [inline tests/test_cmd_init.py]: The description of this test is wrong.

File: tests/test_cmd_init.py [hunk 3]
```diff
@@ -166,14 +175,31 @@ def test_existing_db_error(self):
...
-        self.assertEqual(code2, CODE_DATABASE_ERROR)
+        self.assertEqual(code2, CODE_DATABASE_EXISTS)
 
         # Context added to catch deprecation warnings raised on Python 3
         with warnings.catch_warnings(record=True):
             output = sys.stderr.getvalue().strip()
             self.assertRegexpMatches(output,
                                      DB_EXISTS_ERROR % {'database' : self.name})
 
+        cmd = Init(user=self.kwargs['user'], password=self.kwargs['password'],
```
Comment (sduenas) [inline tests/test_cmd_init.py]: Why do you need to create a new `Init` instance? I think assertions from above cover this case. In my opinion, this was already checked.

File: sortinghat/exceptions.py [hunk 4]
```diff
@@ -90,7 +90,14 @@ class DatabaseError(BaseError):
     """Exception raised when a database error occurs"""
 
     message = "%(error)s (err: %(code)s)"
-    code = CODE_DATABASE_ERROR
+    code = CODE_DATABASE_EXISTS
+
+
+class DatabaseExists(BaseError):
```
Comment (sduenas) [inline sortinghat/exceptions.py]: I miss test cases for this exception. Check `test_exceptions.py` file. Test are straightforward.

**Comment by jgbarah** [comment]
All your comments hopefully addressed.

**Pull request merged**